examples/coap-server.c

```diff
@@ -351,13 +365,22 @@ usage( const char *program, const char *version) {
 
   fprintf( stderr, "%s v%s -- a small CoAP implementation\n"
      "(c) 2010,2011,2015 Olaf Bergmann <bergmann@tzi.org>\n\n"
-     "usage: %s [-A address] [-p port]\n\n"
-     "\t-A address\tinterface address to bind to\n"
-     "\t-g group\tjoin the given multicast group\n"
-     "\t-p port\t\tlisten on specified port\n"
-     "\t-v num\t\tverbosity level (default: 3)\n"
-     "\t-l list\t\tFail to send some datagram specified by a comma separated list of number or number intervals(for debugging only)\n"
-     "\t-l loss%%\t\tRandmoly fail to send datagrams with the specified probability(for debugging only)\n",
+     "Usage: %s [-A address] [-g group] [-p port] [-l loss] [-c certfile]\n"
+     "\t\t[-k key] [-v num]\n\n"
+     "\t-A address\tInterface address to bind to\n"
+     "\t-g group\tJoin the given multicast group\n"
+     "\t-p port\t\tListen on specified port\n"
+     "\t-c certfile\tPEM file containing both CERTIFICATE and PRIVATE KEY\n"
+     "\t       \t\tThis argument requires (D)TLS with PKI to be available\n"
+     "\t-k key\t\tPre-shared key. This argument requires (D)TLS with PSK\n"
+     "\t       \t\tto be available. Default key is secretPSK.\n"
```

Comment:
I am getting more and more uncomfortable with a default key being hard-coded in the server (everybody knows the stories of so-called "IoT appliances" being part of botnets etc. Instead of documenting it here (which is at least a small improvement) we should think about removing the hard-wired default. The key has been in there in the very beginning for some basic tests, but I think that the code has become stable enough that we do not need that loophole anymore.

Instead, it should be documented that the key must not be empty (prohibited by `cmdline_read_key()`).

Reply:
secretPSK is no more (as well as removed from the code)
"\t       \t\tto be available. This cannot be empty if defined.\n"